src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -1067,39 +1106,44 @@ def cancel(self, future):
         """
         Estimates the time for the rest of the acquisition.
-
-        :param roi_idx: (int) index of the current roi
-        :param actual_time_per_roi: (float) average actual time spent for each roi until now
         :return: (float) the estimated time for the rest of the acquisition
         """
-        remaining_rois = (len(self.areas) - roi_idx)
-        if actual_time_per_roi:
-            acquisition_time = actual_time_per_roi * remaining_rois
-        else:
-            autofocus_time = 0
+        autofocus_time = 0
+        self.time_per_task= []
+        if self._use_autofocus:
+            autofocus_time = estimate_autofocus_in_roi_time(self._total_nb_focus_points, self._det, self._focus, self.focus_rng)
+        tiled_time = 0
+        for area in self.areas:
+            tiled_acq_time = estimateTiledAcquisitionTime(
+                                    self.streams, self._stage, area,
+                                    focus_range=self.focus_rng,
+                                    overlap=self._overlap,
+                                    settings_obs=self._settings_obs,
+                                    log_path=self._log_path,
+                                    zlevels=self._zlevels,
+                                    registrar=self._registrar,
+                                    weaver=self._weaver,
+                                    focusing_method=self.focusing_method)
+            tiled_time += tiled_acq_time
+
             if self._use_autofocus:
-                autofocus_time = estimate_autofocus_in_roi_time(self._total_nb_focus_points, self._det)
-            tiled_time = 0
-            for area in self.areas:
-                tiled_time += estimateTiledAcquisitionTime(
-                                        self.streams, self._stage, area,
-                                        focus_range=self.focus_rng,
-                                        overlap=self._overlap,
-                                        settings_obs=self._settings_obs,
-                                        log_path=self._log_path,
-                                        zlevels=self._zlevels,
-                                        registrar=self._registrar,
-                                        weaver=self._weaver,
-                                        focusing_method=self.focusing_method)
-
-            logging.debug(f"Estimated autofocus time: {autofocus_time} s, Tiled acquisition time: {tiled_time} s")
-            acquisition_time = autofocus_time + tiled_time
+                self.time_per_task.extend([autofocus_time, tiled_acq_time])
+            else:
+                self.time_per_task.append(tiled_acq_time)
+
+
+        logging.debug(f"Estimated autofocus time: {autofocus_time * len(self.areas)} s, Tiled acquisition time: {tiled_time} s")
+        acquisition_time = autofocus_time + tiled_time
 
         return acquisition_time
 
+    def _pass_future_progress(self, future, start, end):
```

Comment:
Let's clarify the types for this callback.
```suggestion
    def _pass_future_progress(self, future, start: float, end: float) -> None:
```